util/task.cpp

```diff
@@ -0,0 +1,380 @@
+            throw;
+        // Show warning and print the parsed part of the event
+        fprintf(stderr, "Invalid eSEL format: %s\n", e.what());
+    }
+    printer_.print(event);
+}
+
+std::vector<uint8_t> Task::readFile(const char* path) const
+{
+    std::ifstream file(path, std::ios::in | std::ios::binary);
+    if (!file)
+        throw std::system_error(errno, std::system_category(), path);
+
+    std::vector<uint8_t> content;
+
+    std::copy(std::istreambuf_iterator<char>(file),
+              std::istreambuf_iterator<char>(), std::back_inserter(content));
+
+    return content;
+}
+
+std::vector<uint8_t> Task::removeEcc(const std::vector<uint8_t>& data) const
+{
+    std::vector<uint8_t> result;
+
+    const size_t len = data.size();
+    result.reserve(len - len / 9);
+    size_t i = 0;
+    for (uint8_t n : data)
+    {
+        if (++i == 9)
+            i = 0;
+        else
+            result.push_back(n);
+    }
+
+    return result;
+}
+
+std::vector<uint8_t> Task::readBmcEvent() const
+{
+    std::vector<uint8_t> eselRaw;
+
+    // Read BMC event
+    const std::string eventFile =
+        BmcEventPath + std::string("/") + std::to_string(bmcEventId_);
+    const std::vector<uint8_t> eventData = readFile(eventFile.c_str());
+
+    auto it = std::search(eventData.begin(), eventData.end(), BmcEselToken,
+                          BmcEselToken + sizeof(BmcEselToken));
+    if (it != eventData.end())
+    {
+        // Cut out ESEL property value
+        const uint8_t* eselStart = &*it + sizeof(BmcEselToken);
+        const uint8_t* eselEnd = eselStart + 64;
+        while ((isspace(*eselEnd) || isxdigit(*eselEnd)) &&
+               eselEnd < &*eventData.end())
+            ++eselEnd;
+        // ...and convert it from hex dump
+        eselRaw.reserve((eselEnd - eselStart) / 3 /* hex plus space */);
+        const uint8_t* eselPos = eselStart;
+        while (eselPos < eselEnd)
+        {
+            // Skip spaces
+            while (isspace(*eselPos))
+                ++eselPos;
+            // Convert hex to numeric
+            uint8_t val = 0;
+            for (uint8_t hb = 0; hb <= 1; ++hb)
+            {
+                val <<= 4 * hb;
+                if (*eselPos >= '0' && *eselPos <= '9')
+                    val |= *eselPos - '0';
+                else if (*eselPos >= 'a' && *eselPos <= 'f')
+                    val |= 0x0a + *eselPos - 'a';
+                else if (*eselPos >= 'A' && *eselPos <= 'F')
+                    val |= 0x0a + *eselPos - 'A';
+                ++eselPos;
+            }
+            eselRaw.push_back(val);
+        }
+    }
+
+    if (eselRaw.empty())
+    {
+        throw std::runtime_error(std::string("BMC event with ID ") +
+                                 std::to_string(bmcEventId_) +
+                                 " doesn't contain eSEL entry");
+    }
+
+    return eselRaw;
+}
+
+void Task::printBmcEventList() const
+{
+    DIR* dir = opendir(BmcEventPath);
```

Comment:
https://en.cppreference.com/w/cpp/experimental/fs is also very fashionable

Reply:
It is still "experimental". Let's wait for standard.

Reply:
No, it was applied to C++17.
I've post wrong link. Here is a valid - https://en.cppreference.com/w/cpp/filesystem